src/ua/artcode/udiary/model/Dairy.java

```diff
@@ -1,5 +1,6 @@
 package ua.artcode.udiary.model;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class Dairy {
```

Comment:
To simply filtering logic and using this structure in collections I would override equals and hashcode

Reply:
What fields should be included in it? I mean is it correct to use in hashCode() only i field and in equals - all the rest fields?